lilcom/fixed_math.h

```diff
@@ -283,7 +274,7 @@ void CopyFromScalar64(const Scalar64 *a, int i, Vector64 *y);
 void MulScalar64(const Scalar64 *a, const Scalar64 *b, Scalar64 *y);
 
 /* does: y := a.  Just copies all the elements of the struct. */
-extern inline void CopyScalar64(const Scalar64 *a, Scalar64 *y) {
+static inline void CopyScalar64(const Scalar64 *a, Scalar64 *y) {
```

Comment:
You sure about 'static' here?  I thought when you declared these in a header it should be extern?
I am not so knowledgeable about this though.

Reply:
Hi dan,

About making "inline" static, i just find it work, so i used it. After you asked, I google it and find it's a tricky problem.
Ref: http://gudok.xyz/inline/

Reasons in brief:
1. Inline funtion is linked diffenently in C and C++, C: internal link, C++: external link. (In C, make definition of inline func be extern, will easily lead to the func duplicated defined)
2. In C, "-O0" don't inline the function actually, but "-O3" leads to real "inline" (except the func cannot be inlined). Thus, with "-O3", the inline func don't need to be "static".
3. But with fake inline("-O0"), the compiler ignores inline function body, thus that missing during linking.

Solutions for C:
1. add “static” to inline func
2. or, explicitly instantiate this function once, and make it extern. (It is a inferior solution.)